fastf1/plotting/_plotting.py

```diff
@@ -176,18 +176,23 @@ def driver_color(identifier: str) -> str:
             ratio = fuzz.ratio(identifier, existing_key)
             key_ratios.append((ratio, existing_key))
         key_ratios.sort(reverse=True)
-        if key_ratios[0][0] != 100:
-            _logger.warning(
-                "Correcting invalid user input "
-                 f"'{identifier}' to '{key_ratios[0][1]}'."
 
```

Comment:
I am swapping the order of these two checks because the old logic flow was confusing.

Users will first see the message about correcting their input to a proper team name, then the KeyError would be thrown without explanation.

I moved the KeyError block up so users do not see the attempted conversion and added an explanatory message for the error

Reply:
Yes, that's a good idea